addons/sale_margin/models/sale_order.py

```diff
@@ -1,14 +1,20 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-
 from odoo import api, fields, models
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    margin = fields.Float(compute='_compute_margin', digits='Product Price', store=True)
-    purchase_price = fields.Float(string='Cost', compute="_compute_purchase_price", digits='Product Price', store=True, readonly=False)
+    margin = fields.Float(
+        "Profit", compute='_compute_margin',
+        digits='Product Price', store=True, groups="base.group_user")
+    margin_percent = fields.Float(
+        'Margin', compute='_compute_margin', store=True, groups="base.group_user")
```

Comment:
@dbo-odoo Would it make sense to have a digits on the margins (and is there an appropriate dp for this?) ? Or do we always want the full precision ?

Reply:
I'd prefer having the precision in db, since someone could use the field to compute stuff down the line (e.g. 3rd party module). It's not a really hard requirement though, as at worst someone who wants perfect precision could re-compute it (it's a division, so not the hardest thing), but I see little to gain from specifying a digits here.

Why, you had something in mind?

Reply:
Nope, just asking because while working on this, I was curious to know whether we globally want to precise dp's on Floats or not (and which would be the correct one for percentages, because there is a dedicated dp for discounts, but not for percentage I think).